### ISSUE TITLE
[Deprecation] `Index` vs `Nonce`

### ISSUE THREAD

**Issue by AurevoirXavier** [paritytech/polkadot-sdk#243]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

Why do we need two different names for the same thing?

IMO this adds extra learning effort for developers.

```rs
use xxx::Index;
pub use xxx::Index as Nonce;

impl frame_system::Config {
  type Index = Index;
}

substrate_frame_rpc_system::AccountNonceApi<Block, AccountId, Nonce>;
```

---

I think maybe they are two different things.

Index is to describe the transaction index in the block.

Nonce is the account transactions counter.

And they are `u32`.

But the `pub use as` makes people confused. 

---

```[tasklist]
### Deprecation steps
- [x] ~Soft deprecation and removal from the codebase~
- [x] Update examples and tutorials https://github.com/substrate-developer-hub/substrate-docs/issues/2035
- [x] ~Hard deprecation by adding a deprecate compiler warning~
- [x] ~Deprecation is noted in the release log~
- [x] Removal version is planned -> Next release
- [x] Deprecated feature is removed from the code base https://github.com/paritytech/substrate/pull/14290
- [x] Removal of the feature is noted in the release log
```

**Comment by KiChjang** [comment]
Fixed in paritytech/substrate#14290.

**Comment by juangirini** [comment]
This is still open awaiting for the last step of the deprecation todo list to be completed https://github.com/substrate-developer-hub/substrate-docs/issues/2035
